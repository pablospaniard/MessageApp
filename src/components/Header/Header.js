import React from 'react'
import styled from 'styled-components'

import { Button } from '../'

const StyledOutter = styled.div`
  margin: 20px;
  display: flex;
  flex-flow: row nowrap;
  justify-content: space-between;
  align-items: center;
`

const StyledTopUP = styled.div`
  display: flex;
  justify-content: space-around;
`

const StyledTitle = styled.div`
  font-size: 1.5rem;
  color: ${props => props.theme.greyText};
`

const Header = props => (
  <StyledOutter className="row">
    <StyledTitle className="col-sm-2">Messages</StyledTitle>
    <div className="col-sm-6">
      <Button text={'all messages'} primary />
      <Button text={'received'} primary />
      <Button text={'sent'} disabled />
    </div>
    <StyledTopUP className="col-sm-2">
      <p>balance</p>
      <Button text={'top up'} danger />
    </StyledTopUP>
    <div className="col-sm-2">Account</div>
  </StyledOutter>
)

export default Header
