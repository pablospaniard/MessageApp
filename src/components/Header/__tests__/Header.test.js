import React from 'react'
import { render, cleanup } from 'react-testing-library'
import { Router } from 'react-router-dom'
import { createMemoryHistory } from 'history'

import Header from '../Header'

function renderWithRouter(
  ui,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) {
  return {
    ...render(<Router history={history}>{ui}</Router>),
    history
  }
}

describe('Header should render properly', () => {
  afterEach(cleanup)

  it('should match snapshot', () => {
    const { container } = renderWithRouter(<Header />)
    expect(container.firstChild).toMatchSnapshot('Header_snapshot_1')
  })
})
