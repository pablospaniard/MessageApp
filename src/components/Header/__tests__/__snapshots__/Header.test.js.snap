// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header_snapshot_1 1`] = `
<div
  class="row sc-bZQynM eGVnJm"
>
  <div
    class="col-sm-2 d-none d-sm-block sc-dnqmqq jdnDoH"
  >
    Messages
  </div>
  <div
    class="col-sm-6 col-8"
  >
    <a
      class="sc-htpNat dSZjDA active"
      aria-current="page"
      href="/"
    >
      <button
        class="sc-bdVaJa gjOhON"
      >
        all messages
      </button>
    </a>
    <a
      class="sc-htpNat dSZjDA"
      href="/received"
    >
      <button
        class="sc-bdVaJa gjOhON"
      >
        received
      </button>
    </a>
    <a
      class="sc-htpNat dSZjDA"
      href="/sent"
    >
      <button
        class="sc-bdVaJa gjOhON"
      >
        sent
      </button>
    </a>
    <a
      class="sc-htpNat dSZjDA"
      href="/send-sms"
    >
      <button
        class="sc-bdVaJa gjOhON"
      >
        send SMS
      </button>
    </a>
  </div>
  <div
    class="col-2 sc-gzVnrw jsQWkZ"
  >
    <div>
      <p
        class="sc-htoDjs kmKBQy"
      >
        balance
      </p>
      <p
        style="color: rgb(255, 117, 108);"
      >
        â‚¬ 
      </p>
    </div>
    <button
      class="sc-bdVaJa gjOhON"
    >
      top up
    </button>
  </div>
  <div
    class="col-sm-22 d-none d-sm-block"
  >
    Account
  </div>
</div>
`;
